ledger/sandbox/src/main/scala/com/digitalasset/platform/store/dao/CommandCompletionsTable.scala

```diff
@@ -61,8 +61,10 @@ object CommandCompletionsTable {
           from participant_command_completions
           where
             completion_offset between $startInclusive and $endExclusive and
-            application_id = ${applicationId: String} and
-            submitting_party in (${parties.asInstanceOf[Set[String]]})"""
+            (application_id is null or application_id = ${applicationId: String}) and
```

Comment:
Hm, this spreads out the clause for checkpoints across two terms. How about something along the lines of
```sql
-- checkpoints
(application_id is null and submitting_party is null and command_id is null) 
or
-- rejections or successful completions
(application_id = ${applicationId: String} and submitting_party in (${parties}))
```

Reply:
Sure, make sense (btw now GitHub supports multi-line suggestions, go wild!)

Reply:
I'll stick to `application_id` and `submitting_party` for checkpoints though, `command_id` is not indexed.

Reply:
Addressed by https://github.com/digital-asset/daml/pull/4735/commits/390025ca382a58b34a71246869dcd1b95fb2be05